dvc/repo/metrics/show.py

```diff
@@ -129,6 +129,8 @@ def _read_metrics(self, metrics, branch):
     res = {}
     for out, typ, xpath in metrics:
         assert out.scheme == "local"
+        if not typ:
+            typ = _check_file_type(out.path, typ)
```

Comment:
Currently if no type is specified, dvc shows metrics as a raw text. With this patch, that will no longer work, as it will try to detect type from the extension. I think that is a useful enough behaviour that we could adopt it, but we need to check that if type is set to `raw` it would indeed print it as raw. Could you please check that?

@mroutis We've talked about type detection with a sniffer in one of the recent patches, what are your thoughts on ^ ?

Reply:
Actually, @efiop , I think it is great to rely on the extension, I would expect that as the default behavior.
Let's say that for some weird reason, the user wants a CSV to be interpreted as a `raw` text, they could change the extension to `.txt` or `modify type` it.

The sniffer works awesome to detect _dialects_ for files with columns delimited by a character (tab, colon, comma, etc.), but it will fail for `json` types.

Reply:
@mroutis Thanks! That `.txt` trick works well, but it is more of a hack, since we still need to be able to tell dvc explicitly what type to use. So let's go with `type: raw`(or/and `txt`) as it is a more explicit solution that goes well with what we have now. Judging by the way `dvc metrics show` treats explicit types right now, any unsupported type will just fall back to raw/txt, but we need to make sure that is the case. @cand126 Could you please make sure that `raw`/`txt` types are printed properly, please?